### PR TITLE
Remove confusing alias / config_setting documentation.

### LINKED ISSUE #6449: Documentation/behavior mismatch with alias + config_setting
### Description of the problem / feature request:
The [documentation](https://docs.bazel.build/versions/master/be/general.html#alias) claims that:
> Aliasing only works for "regular" targets. In particular, package_group, config_setting and test_suite rules cannot be aliased.

However, this does not seem to be accurate, at least for `config_setting`. You can create an `alias` with `actual` set to a `config_setting`, and then the `alias` may be used as a condition itself.

Is this a documentation bug or is this not supposed to work (or does it not work in some subtle way that I'll discover down the line)?

The context is that I would like to turn "raw" `config_setting`s  into something more akin to feature flags, where several different possible values for a setting map to a single condition that makes more sense for consumers. Aliasing works for this case although it seems a bit hacky; if there's a better way to achieve the same thing I'd be interested to know.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Full reproducer [here](https://github.com/jfancher/bazel-issues/tree/alias_config_setting). 

### What operating system are you running Bazel on?

macOS 10.13.2

### What's the output of `bazel info release`?

`release 0.18.0`

###  Have you found anything relevant by searching the web?

No

### PR DESCRIPTION
**Pull request by gregestren**
Fixes #6449

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/Alias.java [hunk 1]
```diff
@@ -93,11 +93,6 @@ public Metadata getMetadata() {
   The <code>alias</code> rule creates another name a rule can be referred to as.
 </p>
 
-<p>
-  Aliasing only works for "regular" targets. In particular, <code>package_group</code>,
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/rules/Alias.java]: Are you sure this works for `package_group` and `test_suite`? Is this functionality tested? `test_suite` rules are (unfortunately) special-cased in a number of ways e.g. in `bazel query` and other places expanding them (unfortunately, we have more than one) and I have no idea whether aliases are taken into account there.

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/Alias.java]: For `package_group` and `test_suite` I'm following the thought process described at https://github.com/bazelbuild/bazel/issues/6449#issuecomment-433712899.

Comment (lberki) [reply src/main/java/com/google/devtools/build/lib/rules/Alias.java]: I think you are technically correct, but the fact that `package_group` is not a proper rule is a very obscure piece of information (except for us who have worked on Bazel since forever) and `test_suite` *is* a rule. Why not call this out explicitly?

(btw, `package_group` totally should be a rule. Totally my fault that it's not.)

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/Alias.java]: Call out what explicitly? Like, don't get into the details that `package_group` and `test_suite` are "special" but call them out.. how?

I don't mind adding some simple tests to check that alias works in a standard build use, but I wouldn't be enthused to check all the corner cases with query and who knows where else. :p

**Comment by jin** [comment]
Can this be imported?

File: src/main/java/com/google/devtools/build/lib/rules/Alias.java (hunk 1 above)
Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/Alias.java]: Updated to just remove reference to `config_setting`, which is really the point of the bug. There's no evidence the comments on `package_group` and `test_suite` are confusing people, so let's just leave them be.

**Comment by gregestren** [comment]
> Can this be imported?

With an approval, yes. :)
